src/main/java/org/aludratest/service/jms/impl/JmsActionImpl.java

```diff
@@ -190,7 +190,10 @@ public String sendMessage(FileMessageData fileMessageData, String destinationNam
 			String fileContent = IOUtil.getContentOfURI(fileMessageData.getFileUri());
 			
 			TextMessageData textMessageData = new TextMessageData(fileContent);
-			textMessageData.setProperties(fileMessageData.getProperties());
+			for(Map.Entry<String, Object> entry: fileMessageData.getProperties().entrySet()) {
+				textMessageData.addProperty(entry.getKey(), entry.getValue());
+			}
+			
 			
```

Comment:
See above - when setProperties is changed, you could still use it here. I think this block is a good argument to keep that `setProperties()` method - with the right implementation - to avoid the loop here.